src/rmc-compiler/src/main.rs

```diff
@@ -0,0 +1,227 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+//! This is the main entry point to our compiler driver. This code accepts a few options that
+//! can be used to configure goto-c compilation as well as all other flags supported by rustc.
+//!
+//! Like miri, clippy, and other tools developed on the top of rustc, we rely on the
+//! rustc_private feature and a specific version of rustc.
+
+#![feature(rustc_private, once_cell)]
+extern crate rustc_codegen_ssa;
+extern crate rustc_driver;
+extern crate rustc_session;
+
+use clap::{
+    app_from_crate, crate_authors, crate_description, crate_name, crate_version, App, AppSettings,
+    Arg, ArgMatches,
+};
+use rmc_queries::{QueryDb, UserInput};
+use rustc_driver::{init_env_logger, install_ice_hook, Callbacks, RunCompiler};
+use std::path::PathBuf;
+use std::rc::Rc;
+
+/// This function generates all rustc configurations required by our goto-c codegen.
+fn rustc_gotoc_flags(lib_path: &str) -> Vec<String> {
```

Comment:
Do you have any objection to my moving this into `cargo-rmc` in the future instead of having a `--goto-c` flag to turn them on? That was my original plan. (This is fine for now ofc)

Reply:
Let's talk more about this one. I'm not 100% sure what the solution that you are proposing is.